File: typescript/structural/bridge/remote_device/remote.ts
import {Device} from "./device";

export class Remote {
  device: Device;

  constructor(device: Device) {
    this.device = device
  }

  togglePower() {
    if (this.device.isEnabled()) {
      this.device.disable()
    } else {
      this.device.enable()
    }
    return this.device.isEnabled()
  }

  volumeDown() {
    const volume = this.device.getVolume()
    if (volume >= 10) {
      this.device.setVolume(volume - 10)
    }
    return this.device.getVolume()
  }

  volumeUp() {
    const volume = this.device.getVolume()
    if (volume <= 90) {
      this.device.setVolume(volume + 10)
    }
    return this.device.getVolume()
  }

  channelDown() {
    const channel = this.device.getChannel()
    if (channel > 1) {
      this.device.setChannel(channel - 1)
    }
    return this.device.getChannel()
  }

  channelUp() {
    const channel = this.device.getChannel()
    if (channel < 100) {
      this.device.setChannel(channel + 1)
    }
    return this.device.getChannel()
  }
}

File: dart/structural/bridge/remote_device/bin/remote.dart
import './device.dart';

class Remote {
  final Device device;

  Remote(this.device);

  bool togglePower() {
    if (device.isEnabled()) {
      device.disable();
    } else {
      device.enable();
    }
    return device.isEnabled();
  }

  int volumeDown() {
    int volume = device.getVolume();
    if (volume >= 10) {
      device.setVolume(volume - 10);
    }
    return device.getVolume();
  }

  int volumeUp() {
    int volume = device.getVolume();
    if (volume <= 90) {
      device.setVolume(volume + 10);
    }
    return device.getVolume();
  }

  int channelDown() {
    int channel = device.getChannel();
    if (channel >= 1) {
      device.setChannel(channel - 1);
    }
    return device.getChannel();
  }

  int channelUp() {
    int channel = device.getChannel();
    if (channel <= 99) {
      device.setChannel(channel + 1);
    }
    return device.getChannel();
  }
}
